src/Aws/AbstractApi.php

```diff
@@ -40,6 +53,18 @@ public function __construct(HttpClientInterface $httpClient, $configuration)
 
         $this->httpClient = $httpClient;
         $this->configuration = $configuration;
+        if ($configuration->get('accessKeyId') && $configuration->get('accessKeySecret')) {
```

Comment:
I'd rather implement a `ConfigurationProvider` that return null if the key is missing.